File: sample-videochat-conference-java/src/main/java/com/quickblox/sample/videochat/conference/java/activities/SplashActivity.java
package com.quickblox.sample.videochat.conference.java.activities;

import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.Handler;
import android.widget.TextView;

import com.quickblox.sample.videochat.conference.java.R;
import com.quickblox.sample.videochat.conference.java.utils.ErrorUtils;
import com.quickblox.sample.videochat.conference.java.utils.SharedPrefsHelper;


public class SplashActivity extends BaseActivity {
    private static final int SPLASH_DELAY = 1500;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        fillUI();

        SharedPrefsHelper sharedPrefsHelper = SharedPrefsHelper.getInstance();
        if (sharedPrefsHelper.hasQbUser()) {
            DialogsActivity.start(this);
            finish();
            return;
        }

        new Handler().postDelayed(() -> {
            LoginActivity.start(SplashActivity.this);
            finish();
        }, SPLASH_DELAY);
    }

    private void fillUI() {
        if (actionBar != null) {
            actionBar.hide();
        }
        TextView appNameTextView = findViewById(R.id.text_splash_app_title);
        TextView versionTextView = findViewById(R.id.text_splash_app_version);

        appNameTextView.setText(R.string.app_name);

        try {
            String versionName = getPackageManager().getPackageInfo(getPackageName(), 0).versionName;
            versionTextView.setText(versionName);
        } catch (PackageManager.NameNotFoundException e) {
            ErrorUtils.showSnackbar(findViewById(R.id.layout_root), R.string.error, e, R.string.dlg_ok, null);
        }
    }
}

File: sample-conference-kotlin/app/src/main/java/com/quickblox/sample/conference/kotlin/presentation/screens/splash/SplashActivity.kt
package com.quickblox.sample.conference.kotlin.presentation.screens.splash

import android.Manifest
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.widget.Toast
import com.quickblox.sample.conference.kotlin.R
import com.quickblox.sample.conference.kotlin.databinding.ActivitySplashBinding
import com.quickblox.sample.conference.kotlin.presentation.screens.base.BaseActivity
import com.quickblox.sample.conference.kotlin.presentation.screens.call.CallActivity
import com.quickblox.sample.conference.kotlin.presentation.screens.login.LoginActivity
import com.quickblox.sample.conference.kotlin.presentation.screens.main.MainActivity
import com.vmadalin.easypermissions.EasyPermissions
import dagger.hilt.android.AndroidEntryPoint

/*
 * Created by Injoit in 2021-09-30.
 * Copyright © 2021 Quickblox. All rights reserved.
 */

private const val NOTIFICATIONS_PERMISSIONS_CODE = 333
private const val SPLASH_DELAY = 1500L

@AndroidEntryPoint
class SplashActivity : BaseActivity<SplashViewModel>(SplashViewModel::class.java), EasyPermissions.PermissionCallbacks {
    private lateinit var binding: ActivitySplashBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySplashBinding.inflate(layoutInflater)
        setContentView(binding.root)
        fillVersion()

        checkAndRequestNotificationPermission()

        viewModel.liveData.observe(this) { result ->
            result?.let { (state, data) ->
                when (state) {
                    ViewState.SHOW_LOGIN_SCREEN -> {
                        LoginActivity.start(this)
                        finish()
                    }

                    ViewState.SHOW_CALL_SCREEN -> {
                        CallActivity.start(this)
                        finish()
                    }

                    ViewState.SHOW_MAIN_SCREEN -> {
                        MainActivity.start(this)
                        finish()
                    }

                    ViewState.ERROR -> {
                        Toast.makeText(baseContext, "$data", Toast.LENGTH_SHORT).show()
                    }
                }
            }
        }
    }

    private fun fillVersion() {
        binding.tvVersion.text = packageManager.getPackageInfo(packageName, 0).versionName
    }

    override fun showProgress() {
        // empty
    }

    override fun hideProgress() {
        // empty
    }


    private fun checkAndRequestNotificationPermission() {
        if (EasyPermissions.hasPermissions(this, Manifest.permission.POST_NOTIFICATIONS)) {
            Handler(Looper.getMainLooper()).postDelayed({
                viewModel.run()
            }, SPLASH_DELAY)
        } else {
            requestNotificationPermission()
        }
    }

    private fun requestNotificationPermission() {
        EasyPermissions.requestPermissions(
            host = this,
            rationale = getString(R.string.notification_permission),
            requestCode = NOTIFICATIONS_PERMISSIONS_CODE,
            perms = arrayOf(Manifest.permission.POST_NOTIFICATIONS)
        )
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this)
    }

    override fun onPermissionsDenied(requestCode: Int, perms: List<String>) {
        viewModel.run()
    }

    override fun onPermissionsGranted(requestCode: Int, perms: List<String>) {
        when (requestCode) {
            NOTIFICATIONS_PERMISSIONS_CODE -> {
                viewModel.run()
            }
        }
    }
}